nomad/config.go

```diff
@@ -398,9 +398,9 @@ func DefaultConfig() *Config {
 	// Disable shutdown on removal
 	c.RaftConfig.ShutdownOnRemove = false
 
-	// Enable interoperability with new raft APIs, requires all servers
-	// to be on raft v1 or higher.
-	c.RaftConfig.ProtocolVersion = 2
+	// Enable interoperability with new raft APIs and autopilot features,
+	// requires all servers to be on raft v3 or higher.
```

Comment:
Hm not sure how to interpret this line. I think the old comment said `raft v1 or higher` as defaulting to v2 was safe even if  some servers were v1 as it would downgrade.

I think the intent of the new `raft v3 or higher` is that _autopilot_ requires v3? Servers with raft=v3 set should still be compatible with v2.

Maybe a clearer comment is:

```go
	// Default to Raft v3 to enable new Raft and autopilot features.
	// Compatible with v2 servers.
```